### ISSUE TITLE
Unsatisfactory result.

### ISSUE THREAD

**Issue by GuHuangAI** [LuChengTHU/dpm-solver#23]
I have trained a 1000-step diffusion model, and get fine results using 1000-step reverse process.
This is the original imgs (the two are concatenated):
![ori2](https://user-images.githubusercontent.com/55237825/207287086-8a75bf6a-93d3-450d-84d4-89ccca7b0502.png)
and this is the generated imgs:
![generate2](https://user-images.githubusercontent.com/55237825/207287132-49a541e7-402d-419e-b0f3-96fe60f1a40a.png)

However, when using dpm_solver, i get unsatisfactory or even worse results.
Here is the 20-step img:
![dpm-20](https://user-images.githubusercontent.com/55237825/207288254-de5b33a9-23ee-42f3-86b4-7c2241f49a58.png)

100-step:
![dpm](https://user-images.githubusercontent.com/55237825/207288302-05386ab6-7f22-4378-bb61-de2902e53a81.png)

What happended and what should i do ?

**Comment by LuChengTHU** [comment]
Hi @GuHuangAI ,

Could you please provide a detailed example code for using DPM-Solver? (e.g. which algorithm and which hyperparameters)

**Comment by GuHuangAI** [comment]
> Hi @GuHuangAI ,
> 
> Could you please provide a detailed example code for using DPM-Solver? (e.g. which algorithm and which hyperparameters)

Thanks to your reply.
Actually, i modified the original Unet and designed a mask-controlled diffusion model.
Therefore, my model has an additional input, and i modified the "model_fn" function of DPM_Solver.
My code is like this:
`## 1. Define the noise schedule.
noise_schedule = NoiseScheduleVP(schedule='cosine')

## 2. Convert your discrete-time `model` to the continuous-time
## noise prediction model. Here is an example for a diffusion model
## `model` with the noise prediction type ("noise") .
model_fn = model_wrapper(
    model,
    noise_schedule,
    model_type="noise",  # or "x_start" or "v" or "score"
    # model_kwargs=None,
)

## 3. Define dpm-solver and sample by singlestep DPM-Solver.
## (We recommend singlestep DPM-Solver for unconditional sampling)
## You can adjust the `steps` to balance the computation
## costs and the sample quality.
dpm_solver = DPM_Solver(model_fn, noise_schedule, algorithm_type="dpmsolver")
## Can also try
# dpm_solver = DPM_Solver(model_fn, noise_schedule, algorithm_type="dpmsolver++")

## You can use steps = 10, 12, 15, 20, 25, 50, 100.
## Empirically, we find that steps in [10, 20] can generate quite good samples.
## And steps = 20 can almost converge.
x_T = torch.randn((mask.shape[0], 3, *mask.shape[2:]), device=mask.device)
x_sample = dpm_solver.sample(
    x_T,
    mask,
    steps=20,
    order=2,
    skip_type="time_uniform",
    method="multistep",
)` 
In a wod, i use the mask to generate imgs:
Input:
![mask2](https://user-images.githubusercontent.com/55237825/207302172-0643e919-44aa-452b-86eb-aca8e3556cf4.png)
Output:
![generate2](https://user-images.githubusercontent.com/55237825/207302453-463a6e35-5f3c-4493-ace6-e4618ff2667e.png)

Since my work is used for industrial projects. I'm sorry that i can not share my code.

**Comment by GuHuangAI** [comment]
@LuChengTHU , Thanks for your suggestions. I have tried a lot of combinations of superparametes, unfortunately，they all failed. Maybe my model is  so sensitive that it can only satisfy with much steps. (-. -)

**Comment by LuChengTHU** [comment]
Hi @GuHuangAI ,

Have you tried the original DDIM? Can it provide clean samples? If it can not provide clean samples either, then maybe it is related to your sensitive model. Please follow this instruction to check whether dpmsolver is suitable for your model: https://github.com/LuChengTHU/dpm-solver#suggestions-for-choosing-the-hyperparameters
